model/converter/thrift/zipkin/to_domain.go

```diff
@@ -309,7 +356,10 @@ func (td toDomain) getLogFields(annotation *zipkincore.Annotation) []model.KeyVa
 
 func (td toDomain) getSpanKindTag(annotations []*zipkincore.Annotation) (model.KeyValue, bool) {
 	for _, a := range annotations {
-		if spanKind, ok := coreAnnotations[a.Value]; ok {
+		if spanKind, ok := coreClientAnnotations[a.Value]; ok {
```

Comment:
I have split core annotations to be sure that client kind is always created first. We could keep annotations in one map and then in `transformSpan` check which one has been transformed and create missing one based on that

Reply:
> to be sure that client kind is always created first

why does it matter?

Reply:
Then in `transformSpan` I know that only server span kind has to be added.